pkg/doc/jose/decrypter.go

```diff
@@ -79,22 +74,52 @@ func (jd *JWEDecrypt) Decrypt(jwe *JSONWebEncryption) ([]byte, error) {
 		return nil, fmt.Errorf("jwedecrypt: failed to build encryptedData for Decrypt(): %w", err)
 	}
 
+	authData, err := computeAuthData(protectedHeaders, []byte(jwe.AAD))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(jwe.Recipients) == 1 {
+		authData = []byte(jwe.OrigProtectedHders)
+	}
+
 	return decPrimitive.Decrypt(encryptedData, authData)
 }
 
 func buildEncryptedData(encAlg string, jwe *JSONWebEncryption) ([]byte, error) {
 	var recipients []*subtle.RecipientWrappedKey
 
-	for _, recJWE := range jwe.Recipients {
-		rec, err := convertMarshalledJWKToRecKey(recJWE.Header.EPK)
+	if len(jwe.Recipients) == 1 { // compact serialization: it has only 1 recipient with no headers
```

Comment:
nit: this comment and the one besides the `else` - while not technically wrong - are kind of misleading because there is no JWT serialization here

Reply:
This is not for JWT serialization, it's about JWE serialization. This function (`convertMarshalledJWKToRecKey`) is only a utility function to get a recipient instance, regardless of the serialization type used here.